Creationpane layout

Markvanderuit opened the pull request.

Markvanderuit committed:
Added extra modal view window for gui testing.

Started work on button styling.

Markvanderuit committed:
Created TransitionHelper class to assist in easy adding of transitions to nodes, for common events such as mouse-over, clicks, etc.

And wowie this makes stuff look cool. :D

Markvanderuit committed:
Added double offset transition functions.

Markvanderuit committed:
Improved transition functions to have more secure memory about previous transitions, so they can overlap.

Markvanderuit committed:
Added classes representing stylized buttons and textfields.

Markvanderuit committed:
Some work on fixing issues with the blur, and experimenting with the buttons.

Markvanderuit committed:
Some improvements to the blur.

Markvanderuit committed:
Finally fixed the glitch where blurred image would not clean up properly on every frame.

Markvanderuit committed:
Work on stylized checkboxes.

Markvanderuit committed:
Changes to style elements.
Split into multiple css files.

Markvanderuit committed:
Merge branch 'develop' into css-basics

# Conflicts:
#	src/main/java/gui/root/RootCenterArea.java

Markvanderuit committed:
Continued work on CSS. Also, I have to catch a train, so commit-time.

Markvanderuit committed:
Continued work on checkboxes.

Markvanderuit committed:
Finished checkbox.

Markvanderuit committed:
Merge branch 'develop' into css-basics

Markvanderuit committed:
Layout for creationpane.
Cleaned up somewhat for checkstyle.

Markvanderuit committed:
Merge branch 'develop' into css-basics

Conflicts:
	src/main/java/gui/modal/CreationModalView.java

Markvanderuit committed:
More work on layout of creation view.
Furthermore, cleaned up checkstyle.

Markvanderuit committed:
Removed the mockupview because it was being a @#$!.

bartdejonge1996 commented:
You call this branch css-basics, but htis is no basics anymore. Please keep PR a LOT smaller than this.

bart-de-jonge commented on src/main/java/gui/misc/BlurHelper.java:
```diff
@@ -1,46 +1,60 @@
-    private SnapshotParameters parameters; // Parameters used for writing snapshots (resolution etc)
-    private Bounds bounds; // Bounds of node used for snapshot parameters (x, y, width, height)
-
+    /**
```
Javadoc usually belongs to a method or specific variable. This comment is about multiple variable, so make it a general line or block comment.

bart-de-jonge commented on src/main/java/gui/misc/BlurHelper.java:
```diff
@@ -1,46 +1,60 @@
     private Point2D offset; // Pixel offset if we wish to move blur slightly
 
     /**
+     * Object and scene variables.
```
Same, change javadoc to general line or block comment

Markvanderuit replied:
Done. Running through all other classes now and changing it there as well.

Markvanderuit committed:
Changed some javadoc to block comments.

bart-de-jonge commented on src/main/java/gui/misc/BlurHelper.java:
```diff
@@ -124,4 +142,24 @@ public void processBlurUsingBounds() {
+     * @param watchable the scrollpane to watch.
+     */
+    public void watchScrolling(ScrollPane watchable) {
+        ChangeListener<Number> listener = new ChangeListener<Number>() {
```
maybe extract this into a seperate method and use a lambda, also use a more descriptive name

Markvanderuit replied:
That method is horribly broken and won't be used again. I just removed it.

bart-de-jonge commented on src/main/java/gui/misc/TransitionHelper.java:
```diff
@@ -0,0 +1,322 @@
+import javafx.util.Duration;
+import lombok.Getter;
+
+
```
whitespace

bart-de-jonge commented on src/main/java/gui/misc/TransitionHelper.java:
```diff
@@ -0,0 +1,322 @@
+        transitionStores = new ArrayList<Double>();
+    }
+
+    /**
```
change to regular comment

bart-de-jonge commented on src/main/java/gui/misc/TransitionHelper.java:
```diff
@@ -0,0 +1,322 @@
+                v, true, interpolator);
+
+        // Bind event handlers on mouse enter and exit for this Node.
+        node.addEventHandler(MouseEvent.MOUSE_PRESSED, mouseInHandler);
```
I think inlining the handler initialisation

bart-de-jonge commented on src/main/java/gui/misc/TransitionHelper.java:
```diff
@@ -0,0 +1,322 @@
+     */
+    public <T> void addMouseClickTransition(Property<T> property, int ms,
+                                            T x, T y) {
+        addMouseClickTransition(property, ms, x, y, Interpolator.LINEAR);
```
Is this one-line method necessary?

Ezlo123 replied:
Unused. Removed.

bart-de-jonge commented on src/main/java/gui/misc/TransitionHelper.java:
```diff
@@ -0,0 +1,322 @@
+     */
+    public void addMouseOverTransition(DoubleProperty property, int ms,
+                                       double v) {
+        addMouseOverTransition(property, ms, v, Interpolator.LINEAR);
```
Same story, a method body of one line, is this necessary?

bart-de-jonge commented on src/main/java/gui/misc/TransitionHelper.java:
```diff
@@ -0,0 +1,322 @@
+     */
+    public <T> void addMouseOverTransition(Property<T> property, int ms,
+                                       T x, T y) {
+        addMouseOverTransition(property, ms, x, y, Interpolator.LINEAR);
```
one-line method

bart-de-jonge commented on src/main/java/gui/misc/TransitionHelper.java:
```diff
@@ -0,0 +1,322 @@
+    }
+
+    /**
+     * Event handler creation functions below.
```
change to regular comment

bart-de-jonge commented on src/main/java/gui/modal/CreationModalView.java:
```diff
@@ -1,33 +1,69 @@
-import java.util.ArrayList;
-import java.util.List;
+
 
```
whitespace

bart-de-jonge commented on src/main/java/gui/modal/CreationModalView.java:
```diff
@@ -1,33 +1,69 @@
-    private static final int width = 600;
-    private static final int height = 600;
+    /**
+     * Tweakable styling variables.
```
change to regular comment

bart-de-jonge commented on src/main/java/gui/modal/CreationModalView.java:
```diff
@@ -1,33 +1,69 @@
+    private Point3D titlelabelColor = new Point3D(255, 255, 255);
+
+    /**
+     * Misc variables.
```
change to regualar comment

bart-de-jonge commented on src/main/java/gui/modal/CreationModalView.java:
```diff
@@ -81,73 +129,174 @@ public CreationModalView(RootPane rootPane, int numberOfCamerasInTimeline,
+
+        // Add cancel button
+        cancelButton = new StyledButton("Cancel");
+        cancelButton.setOnMouseReleased(e -> {
```
please add this listener in the controller (if now controller is present for the creationmodalview please make one and move the other createmodelview functionality from toolviewcontroller to the new controller)

Markvanderuit replied:
Not implemented. Rest of team disagrees on this decision.

bart-de-jonge replied:
The team disagrees on putting the validation in the controller, the button pressed listener is agreed to be moved to a controller

Markvanderuit replied:
The team doesn't agree on anything. I'm not touching that part of the code anymore, someone else can move it and get shot.

bart-de-jonge commented on src/main/java/gui/modal/CreationModalView.java:
```diff
@@ -81,73 +129,174 @@ public CreationModalView(RootPane rootPane, int numberOfCamerasInTimeline,
+
+        // Add creation button
+        creationButton = new StyledButton("Create");
+        creationButton.setOnMouseReleased(this::createShot);
```
Add this handler in the controller

bart-de-jonge commented on src/main/java/gui/modal/CreationModalView.java:
```diff
@@ -81,73 +129,174 @@ public CreationModalView(RootPane rootPane, int numberOfCamerasInTimeline,
+        content.setAlignment(Pos.CENTER_LEFT);
+        content.setMinWidth(350.0);
+        content.setPrefWidth(10000);
+        content.setPrefHeight(10000);
```
please make constant of these

bart-de-jonge commented on src/main/java/gui/modal/CreationModalView.java:
```diff
@@ -81,73 +129,174 @@ public CreationModalView(RootPane rootPane, int numberOfCamerasInTimeline,
+        this.buttonPane = new HBox();
+        this.buttonPane.setSpacing(20.0);
+        this.buttonPane.setAlignment(Pos.CENTER_LEFT);
+        this.buttonPane.setPrefHeight(10000);
```
please use constants

bart-de-jonge commented on src/main/java/gui/modal/CreationModalView.java:
```diff
@@ -81,73 +129,174 @@ public CreationModalView(RootPane rootPane, int numberOfCamerasInTimeline,
+        titleLabel.setStyle(topStyle);
+        titleLabel.setAlignment(Pos.CENTER_LEFT);
+        titleLabel.setPadding(new Insets(0, 0, 0, 20));
+        titleLabel.setPrefWidth(10000);
```
please use constants

bart-de-jonge commented on src/main/java/gui/modal/CreationModalView.java:
```diff
@@ -81,73 +129,174 @@ public CreationModalView(RootPane rootPane, int numberOfCamerasInTimeline,
+        this.checkboxPane = new FlowPane();
+        this.checkboxPane.setHgap(20.0);
+        this.checkboxPane.setVgap(20.0);
+        this.checkboxPane.setMinWidth(300.0);
```
please use constants

bart-de-jonge commented on src/main/java/gui/styling/StyledCheckbox.java:
```diff
@@ -0,0 +1,249 @@
+            @Override
+            public void run() {
+                mark.setStyle(mark.getStyle().concat("-fx-background-color: rgb("
+                    + r + ","
```
is it necessary this is on three lines, doesn't add too much to readability

bart-de-jonge commented on src/main/java/gui/styling/StyledCheckbox.java:
```diff
@@ -0,0 +1,249 @@
+            public void run() {
+                box.setStyle(box.getStyle().concat("-fx-background-color: rgb("
+                        + r + ","
+                        + g + ","
```
Check comment above

bart-de-jonge commented on src/main/java/gui/styling/StyledTextfield.java:
```diff
@@ -0,0 +1,113 @@
+    public void setTextfieldColor(int r, int g, int b) {
+        setStyle(getStyle().concat("-fx-background-color: rgb("
+                + r + ","
+                + g + ","
```
See previous comment

bart-de-jonge commented on src/main/java/gui/styling/StyledTextfield.java:
```diff
@@ -0,0 +1,113 @@
+    public void setTextColor(int r, int g, int b) {
+        setStyle(getStyle().concat("-fx-text-fill: rgb("
+                + r + ","
+                + g + ","
```
See previous comment

Markvanderuit committed:
Implemented requested changes. This includes:
- Moving all logic from the CreationModalView to a CreationModalViewController
- Moving all CreationModalView specific logic away from ToolViewController
- Lots of extra tweakables and constants in CreationModalView
- Some unused stuff was removed

Markvanderuit committed:
Merge branch 'develop' into css-basics

Conflicts:
	src/main/java/gui/root/RootCenterArea.java

Markvanderuit committed:
Some checkstyle fixes.

Markvanderuit committed:
Fixed failing build.
